terraform/main.tf

```diff
@@ -103,6 +85,71 @@ module "iot_edge" {
   run_api_facade_null_resource_id = module.keyvault.run_api_facade_null_resource_id
 }
 
+module "keyvault" {
+  source              = "./modules/keyvault"
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = var.location
+  resource_uid        = local.resource_uid
+  app_princ_id        = module.appservice.app_princ_id
+  issuing_ca          = local.issuing_ca
+}
+
+## If preferred to deploy the infrastructure without private endpoints then the below sections can be removed
+
+module "private-endpoint-acr" {
+  source                        = "./modules/private-endpoints/acr"
+  resource_uid                  = local.resource_uid
+  resource_group_name           = azurerm_resource_group.rg.name
+  location                      = var.location
+  vnet_name                     = module.iot_edge.vnet_name
+  vnet_id                       = module.iot_edge.vnet_id
+  acr_id                        = module.acr.acr_id
+  push_docker_null_resource_id  = module.acr.push_docker_null_resource_id
+  push_iotedge_null_resource_id = module.acr.push_iotedge_null_resource_id
+}
+
+module "private-endpoint-appservice" {
+  source              = "./modules/private-endpoints/appservice"
+  resource_uid        = local.resource_uid
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = var.location
+  vnet_name           = module.iot_edge.vnet_name
+  vnet_id             = module.iot_edge.vnet_id
+  app_id              = module.appservice.app_id
+}
+
+module "private-endpoint-bastion" {
+  source              = "./modules/private-endpoints/bastion"
+  resource_uid        = local.resource_uid
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = var.location
+  vnet_name           = module.iot_edge.vnet_name
+  vnet_id             = module.iot_edge.vnet_id
+}
+
+module "private-endpoint-iot-hub" {
```

Comment:
maybe we can add DPS into the name to make it clear that it contains both endpoints